lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -224,16 +224,16 @@ def do_3d_projection(self, renderer):
         '''
         Project the points according to renderer matrix.
         '''
-        xyslist = [
-            proj3d.proj_trans_points(points, renderer.M) for points in
-            self._segments3d]
-        segments_2d = [list(zip(xs, ys)) for (xs, ys, zs) in xyslist]
+        shape = self._segments3d.shape
+        segments3d = np.transpose(self._segments3d, axes=[2, 0, 1])
+        segments3d = np.lib.pad(
+            segments3d, ((0, 1), (0, 0), (0, 0)),
+            'constant', constant_values=1)
+        xys = proj3d.proj_transform_vec(segments3d.reshape(4, -1), renderer.M)
+        xys = np.reshape(xys.T, shape)
+        segments_2d = xys[:, :, 0:2]
         LineCollection.set_segments(self, segments_2d)
-
-        # FIXME
-        minz = 1e9
-        for (xs, ys, zs) in xyslist:
-            minz = min(minz, min(zs))
+        minz = np.min(xys[:, :, 2])
```

Comment:
We are not guaranteed a non-empty array, hence the initialization of minz with 1e9.